src/encord_active/lib/model_predictions/writer.py

```diff
@@ -220,16 +270,48 @@ def __prepare_class_id_lookups(self, custom_map: Optional[Dict[str, int]]):
             for obj in self.project.ontology.objects:
                 self.object_class_id_lookup[obj.feature_node_hash] = len(self.object_class_id_lookup)
 
-        self.classification_class_id_lookup: Dict[str, ClassID] = {}
-        for obj in self.project.ontology.classifications:
-            self.classification_class_id_lookup[obj.feature_node_hash] = len(self.classification_class_id_lookup)
+        self.classification_class_id_lookup: Dict[FrameClassification, ClassID] = {}
+        for classification, attribute, option in iterate_classification_attribute_options(self.project.ontology):
+            key = FrameClassification(
+                classification_hash=classification.feature_node_hash,
+                attribute_hash=attribute.feature_node_hash,
+                option_hash=option.feature_node_hash,
+            )
+            self.classification_class_id_lookup[key] = len(self.classification_class_id_lookup)
```

Comment:
could be nicer a dict comprehension using enumerate maybe?

Reply:
In most cases multiline comprehensions are considered bad, in our case, it would look like

```python
        self.classification_class_id_lookup = {
            FrameClassification(
                classification_hash=classification.feature_node_hash,
                attribute_hash=attribute.feature_node_hash,
                option_hash=option.feature_node_hash,
            ): index
            for index, (classification, attribute, option) in enumerate(
                iterate_classification_attribute_options(self.project.ontology)
            )
        }
```

but I think we can extend the iterator and yield both the hashes object and the items object which will make the comprehension almost a single line and still nice and readable (and a few others).

```python
        self.classification_class_id_lookup = {
            key: index for index, (key, _) in enumerate(iterate_classification_attribute_options(self.project.ontology))
        }
```

Reply:
That'd be great! I remember seeing similar comprehensions in other spots as well, might make all of them simpler